### PR TITLE
Added c and cpp support to sem-version

### LINKED ISSUE #42: Extend sem-version for gcc
Extend sem-version and not use this `sudo update-alternatives --set gcc "/usr/bin/gcc-6"`.

### PR DESCRIPTION
**Pull request by mactsouk**
Closes #42

### CODE REVIEW

**Comment by mactsouk** [comment]
It looks like tests have nothing to do with `sem-version`.

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -12,9 +12,12 @@ blocks:
       - name: Run Manual tests
         commands:
           - checkout
-          - sem-service start mongodb
```
Comment (markoa) [inline .semaphore/semaphore.yml]: I think you should return these lines. If they ever fail in CI it's a bug.

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -12,9 +12,12 @@ blocks:
       - name: Run Manual tests
         commands:
           - checkout
-          - sem-service start mongodb 
-          - sem-service status mongodb 
-          - sem-service stop mongodb 
-          - sem-version php 7.0.32
-          - sem-version ruby 2.5.3
-          - sem-version ruby 2.5.9
+          - . sem-version
```
Comment (markoa) [inline .semaphore/semaphore.yml]: What's the point of this line?

File: sem-version [hunk 3]
```diff
@@ -67,6 +67,21 @@ version::change_python() {
 
   return $?
 }
+
+################################################################################
+# Added by Mihalis Tsoukalos
```
Comment (markoa) [inline sem-version]: Thanks but we have Git so no need for lines like this. :)

File: sem-version [hunk 4]
```diff
@@ -67,6 +67,21 @@ version::change_python() {
...
+################################################################################
+version::change_c() {
+  version::change "change-gcc-version"
+
+  return $?
+}
+
+version::change_cpp() {
+  version::change "change-gcc-version"
```
Comment (markoa) [inline sem-version]: is it `change-gcc-version` for C++ too?

**Comment by markoa** [comment]
> It looks like tests have nothing to do with `sem-version`.

Can you create a test file to sem-version? Just bootstrapping it and having test cases for the gcc functionality you've added will be beneficial.

File: sem-version (hunk 4 above)
Comment (mactsouk) [reply sem-version]: Yes - because both C and C++ are using the same compiler.

File: .semaphore/semaphore.yml (hunk 2 above)
Comment (mactsouk) [reply .semaphore/semaphore.yml]: Just to use the latest version of `sem-version` in the VM. It will be removed once the changes are accepted.

**Pull request merged**